pkg/fetcher/fetcher.go

```diff
@@ -401,6 +406,19 @@ func (u *URLFetcher) IsStatusNotFound() bool {
 	return u.StatusCode == http.StatusNotFound
 }
 
+// IsNonretryableClientError returns true if status code is a nonretryable 4XX error. This includes
+// all 4XX errors except 'request timeout', 'locked', and 'too many requests'. Additionally, it
+// does not return 'not found' or 'unauthorized' errors since those are handled uniquely.
+func (u *URLFetcher) IsNonretryableClientError() bool {
```

Comment:
One more comment:

Maybe we can do something like this:
```
if u.IsNonretryableClientError() {
    if u.IsStatusUnauthorized() {
     }
     if u.IsStatusNotFound() {
    }
    ....
    err = fmt.Errorf("Nonretryable error '%s (%d)': URL %s.", http.StatusText(u.StatusCode), u.StatusCode, url)
    return nil, nil, DoNotRetry{Err: err}
}
```
By doing this, you don't need to exclude `NotFound` and `unauthoized` from `IsNonretryableClientError`.